spotbugs-tests/src/test/java/edu/umd/cs/findbugs/SuppressionMatcherMockedTest.java

```diff
@@ -0,0 +1,69 @@
+package edu.umd.cs.findbugs;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Answers;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class SuppressionMatcherMockedTest {
+
+    @Mock
+    private ClassAnnotation classAnnotation;
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    private ClassWarningSuppressor suppressor;
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    private BugInstance bugInstance;
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    private BugInstance uselessSuppressionBugInstance;
+
+    @Mock
+    private BugReporter bugReporter;
+
+    @BeforeEach
+    void init() {
+        when(suppressor.getClassAnnotation().getTopLevelClass()).thenReturn(classAnnotation);
+        when(bugInstance.getPrimaryClass().getTopLevelClass()).thenReturn(classAnnotation);
+    }
+
+    @Test
+    void shouldMatchBugInstance() {
+        // given
+        when(suppressor.match(bugInstance)).thenReturn(true);
+        SuppressionMatcher matcher = new SuppressionMatcher();
+        // when
+        matcher.addSuppressor(suppressor);
+        // then
+        assertTrue(matcher.match(bugInstance), "Should match BugInstance");
```

Comment:
Error messages should give more detail about the failed value(s) if possible, especially for `assertTrue` (which doesn't give much detail natively)

Reply:
I've looked into this and do not see a practical way for the matcher to return _why_ the instance wasn't matched.
We're using the term matcher for two separate things: 
- the suppression matchers we're using to filter out detected bugs
- the hamcrest matchers we're using in unit tests (these matchers do provide messages)

I think it is fine to leave the test like that, it is short enough so we can reason about it

Reply:
The message could at least specify which suppression matcher is being applied? It could be deduced anyway by tracing the test, but seeing it at a glance is better.

Reply:
The suppressor is mocked here, so there's no real reason for the instance to be suppressed besides the suppressor being mocked to return false.